File: scratch/extract_notebook.js
const fs = require('fs');
const path = require('path');

const notebookPath = "c:\\Users\\renug\\OneDrive\\Desktop\\AI Interior Final Project\\generating-interior-design-dcgan-wgan.ipynb";
const outputPath = "c:\\Users\\renug\\OneDrive\\Desktop\\AI Interior Final Project\\scratch\\extracted_code.py";

try {
  if (!fs.existsSync(notebookPath)) {
    console.error(`Error: Notebook not found at {notebookPath}`);
    process.exit(1);
  }

  const notebook = JSON.parse(fs.readFileSync(notebookPath, 'utf8'));
  let codeCells = [];
  
  if (notebook.cells && Array.isArray(notebook.cells)) {
    notebook.cells.forEach((cell, i) => {
      if (cell.cell_type === 'code') {
        const source = Array.isArray(cell.source) ? cell.source.join('') : (cell.source || '');
        codeCells.push(`# --- Cell ${i} ---\n${source}\n\n`);
      }
    });
  }

  fs.mkdirSync(path.dirname(outputPath), { recursive: true });
  fs.writeFileSync(outputPath, codeCells.join(''), 'utf8');
  console.log(`Extracted ${codeCells.length} code cells to ${outputPath}`);
} catch (err) {
  console.error("Error running extraction script:", err);
  process.exit(1);
}

File: scratch/extract_notebook.py
import json
import os

notebook_path = r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\generating-interior-design-dcgan-wgan.ipynb"
output_path = r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\scratch\extracted_code.py"

if not os.path.exists(notebook_path):
    print(f"Error: Notebook not found at {notebook_path}")
    exit(1)

with open(notebook_path, 'r', encoding='utf-8') as f:
    notebook = json.load(f)

code_cells = []
for i, cell in enumerate(notebook.get('cells', [])):
    if cell.get('cell_type') == 'code':
        source = cell.get('source', [])
        code_cells.append(f"# --- Cell {i} ---\n" + "".join(source) + "\n\n")

os.makedirs(os.path.dirname(output_path), exist_ok=True)
with open(output_path, 'w', encoding='utf-8') as f:
    f.writelines(code_cells)

print(f"Extracted {len(code_cells)} code cells to {output_path}")
